app/presenters/taxon_presenter.rb

```diff
@@ -0,0 +1,66 @@
+class TaxonPresenter
+  GRID = :grid
+  ACCORDION = :accordion
+  LEAF = :leaf
+
+  attr_reader :taxon
+  delegate(
+    :content_id,
+    :title,
+    :description,
+    :base_path,
+    :tagged_content,
+    :grandchildren?,
+    :child_taxons,
+    :most_popular_content,
+    :can_subscribe?,
+    to: :taxon
+  )
+
+  def initialize(taxon)
+    @taxon = taxon
+  end
+
+  def rendering_type
+    return GRID if taxon.grandchildren?
+
+    taxon.children? ? ACCORDION : LEAF
+  end
+
+  def renders_as_accordion?
+    rendering_type == ACCORDION
+  end
+
+  def renders_as_leaf?
+    rendering_type == LEAF
+  end
+
+  def renders_as_grid?
+    rendering_type == GRID
+  end
+
+  def accordion_content
+    return [] unless renders_as_accordion?
```

Comment:
Is there any reason we're returning `[]` rather than raising an exception? Should we ever ask for `accordion_content` if `renders_as_accordion? == false`?